### PR TITLE
Enable`quotes` rule to enforce single quotes

### PR DESCRIPTION
**Pull request by omh1280**
For #5369. Enable the [`quotes`](http://eslint.org/docs/rules/quotes) rule and `--fix` 21 files.

This caught and fixed >90 occurrences of when double quotes were used instead of single quotes. 

I'm not sure if this change will have unexpected consequences. The research I've done says `'` and `"` are interchangeable. All tests pass, and this will enforce this rule going forward.

### CODE REVIEW

**Comment by omh1280** [comment]
A lot of the `.html` files define JSON objects, which require `"` instead of `'`. Should l disable the rule for files in, say, `/Apps/Sandcastle/gallery`?

**Comment by pjcozzi** [comment]
> A lot of the .html files define JSON objects, which require " instead of '. Should l disable the rule for files in, say, /Apps/Sandcastle/gallery?

Probably disable.  The intension is for those to look like JSON and to be easily copy and pasted into JSON, which only allows double quotes.

File: gulpfile.js [hunk 1]
```diff
@@ -1075,14 +1075,14 @@ function createCesiumJs() {
...
-        var assignmentName = "['" + path.basename(file, path.extname(file)) + "']";
+        var assignmentName = '[\'' + path.basename(file, path.extname(file)) + '\']';
         if (moduleId.indexOf('Shaders/') === 0) {
             assignmentName = '._shaders' + assignmentName;
         }
 
         var parameterName = moduleId.replace(nonIdentifierRegexp, '_');
 
-        moduleIds.push("'./" + moduleId + "'");
+        moduleIds.push('\'./' + moduleId + '\'');
```
Comment (emackey) [inline gulpfile.js]: There are times when readability is not improved.

Comment (emackey) [inline gulpfile.js]: I wish there was a rule for "allow double quotes when the quoted string contains at least one single quote..."

Comment (shunter) [inline gulpfile.js]: `"avoidEscape": true` is probably the best option for cases like this.

Comment (omh1280) [inline gulpfile.js]: I'll add that and update.

File: gulpfile.js [hunk 2]
```diff
@@ -1092,7 +1092,7 @@ function createCesiumJs() {
 define([' + moduleIds.join(', ') + '], function(' + parameters.join(', ') + ') {\n\
   \'use strict\';\n\
   var Cesium = {\n\
-    VERSION : "' + version + '",\n\
+    VERSION : \'' + version + '\',\n\
```
Comment (omh1280) [inline gulpfile.js]: This is the one by-hand change I had to make. Otherwise ESLint fails when checking `Source/Cesium.js`

**Comment by omh1280** [comment]
Bump!

**Comment by mramato** [comment]
Thanks @omh1280!

**Pull request merged**